File: src/Web/WebSPA/src/app/core/models/candidate-application-details-dto.ts
import { ApplicationStatus } from './application-list-item-dto';

export interface CandidateApplicationDetailsDto {
  readonly id: string;
  readonly candidateProfileId: string;
  readonly jobId: string;
  readonly status: ApplicationStatus;
  readonly submittedAtUtc: string;
  readonly updatedAtUtc: string;
}

File: src/Services/ApplicationService/Application/DTOs/CandidateApplicationDetailsDto.cs
using ApplicationService.Domain.Enums;

namespace ApplicationService.Application.DTOs;

public sealed record CandidateApplicationDetailsDto(
    Guid Id,
    Guid CandidateProfileId,
    string JobId,
    ApplicationStatus Status,
    DateTimeOffset SubmittedAtUtc,
    DateTimeOffset UpdatedAtUtc);